From cd3346339be05f6a09b78e04b4135ed3397f90b0 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Mon, 15 Jul 2019 15:14:19 +0200
Subject: [PATCH 063/105] ASoC: simple-amplifier: Add mono support

Currently there are inputs and outputs for stereo use cases, so add an
input and output for mono.

Signed-off-by: Luca Weiss <luca@z3ntu.xyz>
---
 sound/soc/codecs/simple-amplifier.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/codecs/simple-amplifier.c b/sound/soc/codecs/simple-amplifier.c
index b30fc1f894e1..0b3fec876490 100644
--- a/sound/soc/codecs/simple-amplifier.c
+++ b/sound/soc/codecs/simple-amplifier.c
@@ -42,20 +42,25 @@ static int drv_event(struct snd_soc_dapm_widget *w,
 static const struct snd_soc_dapm_widget simple_amp_dapm_widgets[] = {
 	SND_SOC_DAPM_INPUT("INL"),
 	SND_SOC_DAPM_INPUT("INR"),
+	SND_SOC_DAPM_INPUT("IN"),
 	SND_SOC_DAPM_OUT_DRV_E("DRV", SND_SOC_NOPM, 0, 0, NULL, 0, drv_event,
 			       (SND_SOC_DAPM_POST_PMU | SND_SOC_DAPM_PRE_PMD)),
 	SND_SOC_DAPM_OUTPUT("OUTL"),
 	SND_SOC_DAPM_OUTPUT("OUTR"),
+	SND_SOC_DAPM_OUTPUT("OUT"),
 	SND_SOC_DAPM_REGULATOR_SUPPLY("VCC", 20, 0),
 };
 
 static const struct snd_soc_dapm_route simple_amp_dapm_routes[] = {
 	{ "DRV", NULL, "INL" },
 	{ "DRV", NULL, "INR" },
+	{ "DRV", NULL, "IN" },
 	{ "OUTL", NULL, "VCC" },
 	{ "OUTR", NULL, "VCC" },
+	{ "OUT", NULL, "VCC" },
 	{ "OUTL", NULL, "DRV" },
 	{ "OUTR", NULL, "DRV" },
+	{ "OUT", NULL, "DRV" },
 };
 
 static const struct snd_soc_component_driver simple_amp_component_driver = {
-- 
2.27.0

